api/src/services/taxonomy-service.ts

```diff
@@ -1,43 +1,94 @@
 import { Client } from '@elastic/elasticsearch';
-import { SearchHit, SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { AggregationsAggregate, SearchHit, SearchRequest, SearchResponse } from '@elastic/elasticsearch/lib/api/types';
 import { getLogger } from '../utils/logger';
 
 const defaultLog = getLogger('services/taxonomy-service');
 
+interface ITaxonomySource {
+  unit_name1: string;
+  unit_name2: string;
+  unit_name3: string;
+  taxon_authority: string;
+  code: string;
+  tty_kingdom: string;
+  tty_name: string;
+  english_name: string;
+  note: string | null;
+  end_date: string | null;
+}
+
+/**
+ * Service for retreiving and processing taxonomic data from Elasticsearch.
+ */
 export class TaxonomyService {
-  private async elasticSearch(searchRequest: SearchRequest) {
+  /**
+   * Performs a query in Elasticsearch based on the given search criteria
+   * @param {SearchRequest} searchRequest The Elastic search request object
+   * @returns {Promise<SearchResponse<ITaxonomySource, Record<string, AggregationsAggregate>> | undefined>}
+   * Promise resolving the search results from Elasticsearch
+   */
+  async _elasticSearch(
+    searchRequest: SearchRequest
+  ): Promise<SearchResponse<ITaxonomySource, Record<string, AggregationsAggregate>> | undefined> {
     try {
       const client = new Client({ node: process.env.ELASTICSEARCH_URL });
       return client.search({
-        index: 'taxonomy',
+        index: process.env.ELASTICSEARCH_TAXONOMY_INDEX,
         ...searchRequest
       });
     } catch (error) {
       defaultLog.debug({ label: 'elasticSearch', message: 'error', error });
     }
   }
 
-  private sanitizeSpeciesData = (data: SearchHit<any>[]) => {
-    return data.map((item) => {
-      const label = [
-        item._source.code,
-        [
-          [item._source.tty_kingdom, item._source.tty_name].filter(Boolean).join(' '),
-          [item._source.unit_name1, item._source.unit_name2, item._source.unit_name3].filter(Boolean).join(' '),
-          item._source.english_name
+  /**
+   * Determines if the source object from a taxonomy search query is valid (namely, that it has not expired).
+   * @param source The source object from the search query
+   * @returns {boolean} `true` if the the `end_date` property is not defined or if the date hasn't passed, `false` otherwise.
+   */
+  _isValidTaxonomySource = (source: ITaxonomySource | undefined): boolean => {
+    if (source?.end_date) {
+      return new Date() < new Date(source.end_date);
+    }
+
+    return true;
+  };
+
+  /**
+   * Sanitizes species data retrieved from Elasticsearch.
+   * @param {SearchHit<ITaxonomySource>[]} data The data response fromEelasticsearch
+   * @returns {{ id: string, label: string }} An ID and label pair for each taxonomic code
+   */
```

Comment:
Though, I might suggest using that Document This extension, which will populate the base info for you, if you're not already.

The return here is showing an object, rather than an array of objects.